fs_storage/models/fs_storage.py

```diff
@@ -268,14 +284,29 @@ def _compute_options_properties(self) -> None:
     def _get_marker_file_name(self):
         return ".odoo_fs_storage_%s.marker" % self.id
 
-    def _check_connection(self, fs):
+    def _marker_file_check_connection(self, fs):
         marker_file_name = self._get_marker_file_name()
         try:
             fs.info(marker_file_name)
         except FileNotFoundError:
             fs.touch(marker_file_name)
+
+    def _ls_check_connection(self, fs):
+        fs.ls("", detail=False)
+
+    def _check_connection_with_method(self, fs, check_connection_method):
+        if check_connection_method == "marker_file":
+            self._marker_file_check_connection(fs)
+        elif check_connection_method == "ls":
+            self._ls_check_connection(fs)
         return True
 
+    def _check_connection(self, fs):
+        check_connection_method = self.check_connection_method or self.env.context.get(
+            "force_connection_method", ""
+        )
+        return self._check_connection_with_method(fs, check_connection_method)
```

Comment:
IMP We don't need to pass args via context
```suggestion
        return self._check_connection_with_method(fs, self.check_connection_method)
```

Reply:
The same method is used when connection is checked from wizard, for a fs.storage with empty `check_connection_method`

Reply:
We don't need to use the context to pass args from wizard. We can directly call the `_check_connection_with_method` from the wizard.

Reply:
The call on `_test_config` instead of `_check_connection_with_method` from the wizard is there to use the same notification mechanism.